packages/common/src/i18n/format_date.ts

```diff
@@ -101,6 +101,37 @@ export function formatDate(
   return text;
 }
 
+/**
+ * Return a date with full year instead of converting it to 19{yearProvided}.
+ * See: https://github.com/angular/angular/issues/40377
+ *
+ * We cannot use `new Date(year, month, date)` because it maps years between 0 and 99 to 1900-1999.
+ *
+ * Note that this function returns a Date object whose time is midnight in the current locale's
+ * timezone. In the future we might want to change this to be midnight in UTC, but this would be a
+ * considerable breaking change.
+ */
+function createDate(year: number, month: number, date: number): Date {
+  // The `newDate` is set to midnight (UTC) on January 1st 1970.
+  // In PST this will be December 31st 1969 at 4pm.
+  // In GMT this will be January 1st 1970 at 1am.
+  // (Note that they even have different years, dates and months.)
+  const newDate = new Date(0);
+
+  // `setFullYear()` allows years like 1901 to be set correctly. This function does not
```

Comment:
@petebacondarwin Because 1901 can be set correctly using `new Date(0)`, can't be ?
```suggestion
  // `setFullYear()` allows years like 0001 to be set correctly. This function does not
```

Reply:
Doh!! Good spot @abarghoud  :-)